pancancer_evaluation/utilities/analysis_utilities.py

```diff
@@ -396,6 +434,39 @@ def generate_nonzero_coefficients_lasso_range(results_dir, gene=None):
             yield gene_name, cancer_type, seed, lasso_param, coefs
 
 
+def generate_nonzero_coefficients_lasso_range_msi(results_dir):
+    """Generate coefficients for lasso range experiments.
+
+
+    Arguments
+    ---------
+    results_dir (str): directory to look in for results, subdirectories should
+                       be experiments for individual genes
+
+    Yields
+    ------
+    identifier (str): identifier for given coefficients
+    lasso_param (str): lasso regularization parameter
+    coefs (dict): list of nonzero coefficients for each fold of CV, for the
+                  given identifier
+    """
+    coefs = {}
+    all_features = None
+    for coefs_file in os.listdir(results_dir):
+        if coefs_file[0] == '.': continue
+        if 'signal' not in coefs_file: continue
+        if 'coefficients' not in coefs_file: continue
+        cancer_type = coefs_file.split('_')[1]
+        seed = int(coefs_file.split('_')[-4].replace('s', ''))
+        lasso_param = coefs_file.split('_')[-2].replace('c', '')
```

Comment:
Are you using negative indexing here because there's an unknown number of intermediate columns?

Reply:
Yeah, basically. When I was doing the feature selection experiments a while ago, there were a few feature selection methods (e.g. `pancan_f_test`, `median_f_test`) that had underscores in them, so I switched to negative indexing since that made the number of intermediate columns from the beginning of the filename unknown.

Using underscores in the feature selection methods probably wasn't the best idea in retrospect (I could have used dashes or periods or something else) but I don't think I realized that until I had run a bunch of computationally expensive experiments so I didn't want to change.

I'll make a note of this in the code, although it's not that important for this set of experiments since I'm not varying the feature selection method at all.